src/vs/workbench/contrib/extensions/electron-sandbox/remoteExtensionsInit.ts

```diff
@@ -62,49 +64,22 @@ export class InstallRemoteExtensionsContribution implements IWorkbenchContributi
 			return;
 		}
 
-		this.logService.info(`Evaluating '${settingValue.length}' default remote extensions`);
-
-		const galleryExtensions = await this.extensionGalleryService.getExtensions(settingValue.map((id) => ({ id })), CancellationToken.None);
-		const alreadyInstalledInRemote = await this.extensionManagementServerService.remoteExtensionManagementServer.extensionManagementService.getInstalled(ExtensionType.User);
 		const alreadyInstalledLocally = await this.extensionManagementServerService.localExtensionManagementServer.extensionManagementService.getInstalled(ExtensionType.User);
+		const alreadyInstalledRemotely = await this.extensionManagementServerService.remoteExtensionManagementServer.extensionManagementService.getInstalled(ExtensionType.User);
+		const extensionsToInstall = settingValue
+			.filter(id =>
+				alreadyInstalledLocally.some(ext => areSameExtensions(ext.identifier, { id }))
+				&& !alreadyInstalledRemotely.some(ext => areSameExtensions(ext.identifier, { id }))
```

Comment:
I kept the `alreadyInstalledRemotely` check. Why would we want to do the extra marketplace query, etc if the extension is already present?

Also, the behavior of honoring an existing install makes sense (maybe a user needs an earlier extension version on a single remote for _some reason_)

Reply:
Makes sense